Source/WebCore/platform/network/curl/OpenSSLHelper.cpp

```diff
@@ -301,19 +301,76 @@ static void getSubjectAltName(const X509* x509, Vector<String>& dnsNames, Vector
             if (value->d.iPAddress->length == 4)
                 ipAddresses.append(makeString(data[0], ".", data[1], ".", data[2], ".", data[3]));
             else if (value->d.iPAddress->length == 16) {
-                StringBuilder ipAddress;
-                for (int i = 0; i < 8; i++) {
-                    ipAddress.append(makeString(hex(data[0] << 8 | data[1], 4)));
-                    if (i != 7)
-                        ipAddress.append(":");
-                    data += 2;
-                }
-                ipAddresses.append(ipAddress.toString());
+                Span<uint8_t, 16> dataSpan { data, 16 };
+                ipAddresses.append(canonicalizeIPv6Address(dataSpan));
             }
         }
     }
 }
 
+String canonicalizeIPv6Address(Span<uint8_t, 16> data)
+{
+    bool compressCurrentSection = true;
+    String ipAddressSections[8];
+    String oldSection;
+    StringBuilder newSection;
+    size_t maxZeros = 0;
+    size_t unset = -1;
+    size_t rstart = unset;
```

Comment:
Can these be given better names? "rstart" and "rend" doesn't really describe what they're used for.
Also, I think it would be better if they were std::optional<size_t> instead of using -1